more_itertools/more.py

```diff
@@ -4228,14 +4229,14 @@ def is_scalar(obj):
     if not objects:
         return
 
-    iterables_count = sum(1 for obj in objects if not is_scalar(obj))
+    iterables = list(filterfalse(is_scalar, objects))
+    iterables_count = ilen(iterables)
```

Comment:
It's a list, just use `len`. And I don't think you need the count in the variable. You only use it once or twice, the first time you can just check the list instead, and the second time you can just call `len` there.

Reply:
> It's a list, just use `len`.

My bad, I had `filterfalse` in mind while adding `ilen`.

> ...and the second time you can just call len there.

~~~python
elif markers == iterables_count:
~~~

Should I use `len` here? It would be more performant if I check the length once and not every step of for-loop

Reply:
> Should I use `len` here?

Yes.

> It would be more performant if I check the length once and not every step of for-loop

No. You get there at most once.

Reply:
> You get there at most once.

Am I missing something?

<img width="640" alt="Left side: code with debug print where length eval would happen, right side: run of proposed code with 10 debug prints" src="https://github.com/more-itertools/more-itertools/assets/47028153/5464541e-5755-4736-b110-d401028c51eb">

Reply:
Your debug message is not true. You only get to the `len` call if the `if` condition is false. Prepend the print to the `elif` condition instead: `elif print(...) or condition:`.

Reply:
You are right. If `if` statement failed, then for-loop cancels in both `elif` or `else` branches. Thanks for notice!